### PR TITLE
Improved error message when downloading a file fails #9826

### LINKED ISSUE #9826: Improve error message when downloading a file fails
Copy the following bibtex code:

```bibtex
@Article{levy_organising_2023,
  author       = {Levy, David L.},
  date         = {2023-04},
  journaltitle = {Environmental Politics},
  title        = {Organising responses to climate change: the politics of mitigation, adaptation and suffering},
  doi          = {10.1080/09644016.2023.2204045},
  issn         = {0964-4016},
  number       = {0},
  pages        = {1--3},
  url          = {https://doi.org/10.1080/09644016.2023.2204045},
  urldate      = {2023-05-01},
  volume       = {0},
  file         = {Full Text PDF:https\://www.tandfonline.com/doi/pdf/10.1080/09644016.2023.2204045:application/pdf},
  publisher    = {Routledge},
  shorttitle   = {Organising responses to climate change},
}

```

Paste it on the maintable in a new library 
See the error message: 403 forbidden

Context:
  It's a known issue that sometimes the browser extension shows an error page although the import was successful (and there is an issue already for this).

However, the error dialog on the JabRef side needs improvement: explain what 403 means and suggest solutions (not everyone knows this), and also report the url it tried to download; finally, it should proceed with the import and save the file url just as an linked url in the file field.

Solution: 

1. Improve the error message in https://github.com/JabRef/jabref/blob/e34cd37ea3af7f8997bb1eadf1640431cb471c20/src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java#L475-L478

Include the URL in the error message

For a reference/similar handling, see:  
https://github.com/JabRef/jabref/blob/e34cd37ea3af7f8997bb1eadf1640431cb471c20/src/main/java/org/jabref/gui/EntryTypeViewModel.java#L154-L162

3. Store the url in the file field of the corresponding entry

_Originally posted by @tobiasdiez in https://github.com/JabRef/jabref/issues/9816#issuecomment-1529439931_

### PR DESCRIPTION
**Pull request by yerraga**
Fixes #9826 

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by koppor** [comment]
@yerraga Thank you for the contribution. Can you please fix checkstyle? See https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html for details. Please also use `{}` at the logger (instead of string concatenation)

I think, the whole thing should be tested. Can you dive into WireMock to setup mocking the remote servers - so that each exception path in LinkedFileViewModel is tested? https://wiremock.org/docs/junit-jupiter/

**Comment by ThiloteE** [comment]
To ease organizational workflows I have linked the pull-request to the issue with syntax as described in https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

> <h2 id="linking-a-pull-request-to-an-issue-using-a-keyword">Linking a pull request to an issue using a keyword</h2>
> <p>You can link a pull request to an issue by using a supported keyword 
> in the pull request's description or in a commit message. The pull 
> request <strong>must be</strong> on the default branch.</p>
> <ul>
> <li>close</li>
> <li>closes</li>
> <li>closed</li>
> <li>fix</li>
> <li>fixes</li>
> <li>fixed</li>
> <li>resolve</li>
> <li>resolves</li>
> <li>resolved</li>
> </ul>
> <p>If you use a keyword to reference a pull request comment in another 
> pull request, the pull requests will be linked. Merging the referencing 
> pull request also closes the referenced pull request.</p>
> <p>The syntax for closing keywords depends on whether the issue is in the same repository as the pull request.</p>

File: src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java [hunk 1]
```diff
@@ -473,9 +475,31 @@ public void download() {
...
+                        dialogService.showErrorDialogAndWait(Localization.lang("Error downloading from URL"),
+                                "HTTP Error " + statusCode + ": Not Found. The requested resource could not be found.\nURL: " + urlDownload + "\nPlease check the URL and try again.");
+                    } else {
+                        dialogService.showErrorDialogAndWait(Localization.lang("Error downloading from URL"),
+                                "HTTP Error " + statusCode + ": " + clientException.getMessage() + "\nURL: " + urlDownload + "\nPlease check the URL and try again.");
+                    }
+                } else if (ex instanceof FetcherServerException) {
+                    FetcherServerException serverException = (FetcherServerException) ex;
+                    int statusCode = serverException.getStatusCode();
+                    dialogService.showErrorDialogAndWait(Localization.lang("Error downloading from URL"),
```
Comment (koppor) [inline src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java:495]: The second part is also shown to the user
. Thus, that needs to be localized too. It is OK for me to include the server error here.

**Review by koppor — CHANGES_REQUESTED** [review]

**Comment by Siedlerchr** [comment]
What is the status here? Are you planning to fix the compile errors to get it working?

**Comment by calixtus** [comment]
Closing this issue due to inactivity :zzz: 
Please reopen a PR if you intend to continue on this issue. We really liked your idea and draft.

**Pull request closed without merging**